### ISSUE TITLE
[DuckDB] Support INSERT INTO ... WITH

### ISSUE THREAD

**Issue by plabadens** [sqlfluff/sqlfluff#7283]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Similar to #6649, sqlfluff is giving parsing errors when inserting into a table using a query with CTEs.
For example:

```sql
INSERT INTO tbl BY NAME (WITH ... );
INSERT INTO tbl BY POSITION (WITH ... );
INSERT INTO tbl  (WITH ... );
```

### Expected Behaviour

These statements should at least be parsed correctly.

### Observed Behaviour

Trying to lint any of these statements returns a parsing error.

### How to reproduce

Here is a minimal example script: [duckdb-insert-with.sql](https://github.com/user-attachments/files/23544409/duckdb-insert-with.sql)

```sh
$ duckdb -f duckdb-insert-with.sql
$ sqlfluff lint -d duckdb-insert-with.sql
== [duckdb-insert-with.sql] FAIL
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section: 'INSERT
                       | INTO Cities (CityName, CityId)\nWI...'
WARNING: Parsing errors found and dialect is set to 'duckdb'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Dialect

DuckDB

### Version

```sh
$ sqlfluff --version
sqlfluff, version 3.5.0

$ python --version
Python 3.13.8
```

### Configuration

From `pyproject.toml`:

```toml
[tool.sqlfluff.core]
dialect = "duckdb"
exclude_rules = ["ST07"]
large_file_skip_byte_limit = 50000

[tool.sqlfluff.aliasing]
length.min_alias_length = 4

[tool.sqlfluff.rules.references]
special_chars.quoted_identifiers_policy = "column_aliases"

[tool.sqlfluff.rules.capitalisation]
keywords.capitalisation_policy = "upper"
identifiers.extended_capitalisation_policy = "pascal"
functions.extended_capitalisation_policy = "lower"
literals.capitalisation_policy = "lower"
types.extended_capitalisation_policy = "upper"

```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
